contracts/external/traders/LimitOrders.sol

```diff
@@ -0,0 +1,583 @@
+        "uint256 takerMarket,",
+        "uint256 makerAmount,",
+        "uint256 takerAmount,",
+        "address makerAccountOwner,",
+        "uint256 makerAccountNumber,",
+        "address takerAccountOwner,",
+        "uint256 takerAccountNumber,",
+        "uint256 expiration,",
+        "uint256 salt",
+        ")"
+    ));
+
+    // Number of bytes in an Order struct
+    uint256 constant private NUM_ORDER_BYTES = 320;
+
+    // Number of bytes in a typed signature
+    uint256 constant private NUM_SIGNATURE_BYTES = 66;
+
+    // Number of bytes in a CallFunctionData struct
+    uint256 constant private NUM_CALLFUNCTIONDATA_BYTES = 64;
+
+    // ============ Enums ============
+
+    enum OrderStatus {
+        Null,
+        Approved,
+        Canceled
+    }
+
+    enum CallFunctionType {
+        Approve,
+        Cancel
+    }
+
+    // ============ Structs ============
+
+    struct Order {
+        uint256 makerMarket;
+        uint256 takerMarket;
+        uint256 makerAmount;
+        uint256 takerAmount;
+        address makerAccountOwner;
+        uint256 makerAccountNumber;
+        address takerAccountOwner;
+        uint256 takerAccountNumber;
+        uint256 expiration;
+        uint256 salt;
+        bytes32 orderHash;
+    }
+
+    struct CallFunctionData {
+        CallFunctionType callType;
+        bytes32 orderHash;
+    }
+
+    // ============ Events ============
+
+    event LogOrderCanceled(
+        address canceler,
+        bytes32 orderHash
+    );
+
+    event LogOrderApproved(
+        address approver,
+        bytes32 orderHash
+    );
+
+    event LogOrderTaken(
+        bytes32 orderHash,
+        uint256 fillAmount
+    );
+
+    // ============ Immutable Storage ============
+
+    // Hash of the EIP712 Domain Separator data
+    bytes32 public EIP712_DOMAIN_HASH;
+
+    // ============ Mutable Storage ============
+
+    // order hash => filled amount (maker)
+    mapping (bytes32 => uint256) public g_filled;
+
+    // signer => order hash => status
+    mapping (address => mapping (bytes32 => OrderStatus)) public g_status;
+
+    // ============ Constructor ============
+
+    constructor (
+        address soloMargin,
+        uint256 chainId
+    )
+        public
+        OnlySolo(soloMargin)
+    {
+        /* solium-disable-next-line indentation */
+        EIP712_DOMAIN_HASH = keccak256(abi.encode(
+            EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
+            keccak256(bytes(EIP712_DOMAIN_NAME)),
+            keccak256(bytes(EIP712_DOMAIN_VERSION)),
+            chainId,
+            address(this)
+        ));
+    }
+
+    // ============ External Functions ============
+
+    /**
+     * Cancels an orderHash from msg.sender. Cannot already be canceled.
+     *
+     * @param  orderHash  The hash of the order to cancel
+     */
+    function cancelOrder(
+        bytes32 orderHash
+    )
+        external
+    {
+        cancelOrderInternal(msg.sender, orderHash);
```

Comment:
I don't see any sort of authorization on this function (unless I'm missing something). How are you going to ensure that only the maker account owner of the order can cancel?

Reply:
Oh wait, I see I think, it just adds a cancel indexed off of the sender address. So you can cancel an order hash you don't own but it won't do anything